ext/spl/spl_directory.c

```diff
@@ -2047,13 +2047,7 @@ PHP_METHOD(SplFileObject, __construct)
 	size_t path_len;
 	zend_error_handling error_handling;
 
-	intern->u.file.open_mode = ZSTR_CHAR('r');
-
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "P|Sbr!",
-			&intern->file_name, &open_mode,
-			&use_include_path, &intern->u.file.zcontext) == FAILURE) {
-		intern->u.file.open_mode = NULL;
-		intern->file_name = NULL;
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "P|Sbr!", &intern->file_name, &open_mode, &use_include_path, &intern->u.file.zcontext) == FAILURE) {
```

Comment:
`intern->file_name` could be left with a dangling pointer if a later parameter fails to parse. This isn't currently an issue because `new` will destroy the object, and `ReflectionClass::newInstanceWithoutConstructor()` doesn't actually allow calling `__construct` (prevented by `spl_filesystem_object_get_method_check()`). Nonetheless, it might be preferable to store it to a variable and copy it to `intern` when parsing is successful.

`&intern->u.file.zcontext` doesn't really apply, since it's the last parameter, but for consistency it might still make sense.

Reply:
I've rewritten it so that it takes dedicated stack variables. However, the fact that `intern->file_name` is not working on a copy seems quite dubious.